src/nanoarrow/array_inline.h

```diff
@@ -445,6 +445,179 @@ static inline ArrowErrorCode ArrowArrayFinishElement(struct ArrowArray* array) {
   return NANOARROW_OK;
 }
 
+static inline int8_t ArrowArrayViewIsNull(struct ArrowArrayView* array_view, int64_t i) {
+  const uint8_t* validity_buffer = array_view->buffer_views[0].data.as_uint8;
+  i += array_view->array->offset;
+  switch (array_view->storage_type) {
+    case NANOARROW_TYPE_NA:
+      return 0x01;
+    case NANOARROW_TYPE_DENSE_UNION:
+    case NANOARROW_TYPE_SPARSE_UNION:
+      // Not supported yet
+      return 0xff;
```

Comment:
Arguably it's technically "never null"

Reply:
Right, but I'd say conceptually, the union array itself is always valid.

Reply:
Looking at C++, since type codes are buffer 1, then buffer 0 is nullptr and so the array will consider every slot valid

Reply:
Fair - I guess the Arrow union array expects you to do that manually. In that case leaving this as a TODO is fine.